pkg/generator/vector/sources_test.go

```diff
@@ -0,0 +1,186 @@
+		Entry("Only Application", ConfGenerateTest{
+			CLFSpec: logging.ClusterLogForwarderSpec{
+				Pipelines: []logging.PipelineSpec{
+					{
+						InputRefs: []string{
+							logging.InputNameApplication,
+						},
+						OutputRefs: []string{logging.OutputNameDefault},
+						Name:       "pipeline",
+					},
+				},
+			},
+			ExpectedConf: `
+# Logs from containers (including openshift containers)
+[sources.container_logs]
+  auto_partial_merge = true
+  exclude_paths_glob_patterns = ["/var/log/containers/vector-*_openshift-logging_*.log", "/var/log/containers/elasticsearch-*_openshift-logging_*.log", "/var/log/containers/kibana-*_openshift-logging_*.log"]
+`,
+		}),
+		Entry("Only Infrastructure", ConfGenerateTest{
+			CLFSpec: logging.ClusterLogForwarderSpec{
+				Pipelines: []logging.PipelineSpec{
+					{
+						InputRefs: []string{
+							logging.InputNameInfrastructure,
+						},
+						OutputRefs: []string{logging.OutputNameDefault},
+						Name:       "pipeline",
+					},
+				},
+			},
+			ExpectedConf: `
+# Logs from containers (including openshift containers)
+[sources.container_logs]
+  auto_partial_merge = true
+  exclude_paths_glob_patterns = ["/var/log/containers/vector-*_openshift-logging_*.log", "/var/log/containers/elasticsearch-*_openshift-logging_*.log", "/var/log/containers/kibana-*_openshift-logging_*.log"]
+
+[sources.journal_logs]
+  type = "journald"
+`,
+		}),
+		Entry("Only Audit", ConfGenerateTest{
+			CLFSpec: logging.ClusterLogForwarderSpec{
+				Pipelines: []logging.PipelineSpec{
+					{
+						InputRefs: []string{
+							logging.InputNameAudit,
+						},
+						OutputRefs: []string{logging.OutputNameDefault},
+						Name:       "pipeline",
+					},
+				},
+			},
+			ExpectedConf: `
+# Logs from host audit
+[sources.host_audit_logs]
+  type = "file"
+  ignore_older_secs = 600
+  include = ["/var/log/audit/audit.log"]
+
+# Logs from kubernetes audit
+[sources.k8s_audit_logs]
+  type = "file"
+  ignore_older_secs = 600
+  include = ["/var/log/kube-apiserver/audit.log"]
+
+# Logs from openshift audit
+[sources.openshift_audit_logs]
+  type = "file"
+  ignore_older_secs = 600
+  include = ["/var/log/oauth-apiserver.audit.log"]
+`,
+		}),
+		Entry("All Log Sources", ConfGenerateTest{
+			CLFSpec: logging.ClusterLogForwarderSpec{
+				Pipelines: []logging.PipelineSpec{
+					{
+						InputRefs: []string{
+							logging.InputNameApplication,
+							logging.InputNameInfrastructure,
+							logging.InputNameAudit,
+						},
+						OutputRefs: []string{logging.OutputNameDefault},
+						Name:       "pipeline",
+					},
+				},
+			},
+			ExpectedConf: `
+# Logs from containers (including openshift containers)
+[sources.container_logs]
+  auto_partial_merge = true
+  exclude_paths_glob_patterns = ["/var/log/containers/vector-*_openshift-logging_*.log", "/var/log/containers/elasticsearch-*_openshift-logging_*.log", "/var/log/containers/kibana-*_openshift-logging_*.log"]
```

Comment:
We'll need to eval how to make this useful since vector follows `/var/log/pods`

Reply:
i didnt get you, container log path remains same for both vector and fluent

Reply:
The container log path is different AFAIK between the two collectors.  We  configure fluent to use `/var/log/containers` but vector uses `/var/log/pods`.  This means the exclude pattern would change because the file name changes the order of podname and namespace.